### DISCUSSION TITLE
Build failed while using apk

### DISCUSSION THREAD

**Discussion by gunaNeelamegam** [flet-dev/flet#4347] [Q&A]
### Duplicate Check

- [X] I have searched the [opened issues](https://github.com/flet-dev/flet/issues) and there are no duplicates

### Describe the bug

Trying to build apk getting fail.

### Code sample

<details open><summary>Code</summary>

```shell
venv-py3.12) guna@guna:~/Guna/projects/somthing$ flet build apk --v
[09:15:06] Additional Flutter dependencies: {'flet_appwrite': {'path': '/home/guna/Guna/projects/somthing/flet-build-template/custom_packages/flet_appwrite'}}                         
[09:15:08] Unable to create file 'android/app/src/main/AndroidManifest.xml'. Error message: 'dict object' has no attribute 'android_meta_data'.       
   
```

</details>

### To reproduce

flet --version
0.25.0.dev3526
python version: 3.12.3

### Screenshots / Videos

<details open>
<summary>Captures</summary>

[Upload media here]

</details>

### Operating System

Linux

### Operating system details

linux 

### Flet version

0.25.0.dev3526

### Regression

Yes, it used to work in a previous Flet version (please specify the version in additional details)

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

**Comment by MominIqbal-1234** [comment]
i Try to many way to build apk but its not working 
``` `flutter` command is not available in PATH. Install Flutter SDK.  ```

first i try to build my ubuntu not work same error so now i try in google cloab using this command but same error 
``` `flutter` command is not available in PATH. Install Flutter SDK.  ```

```

# Step 1: Install Java, Android SDK, and NDK
!apt-get update -y
!apt-get install -y openjdk-11-jdk

# Download and set up Android SDK and NDK
!wget https://dl.google.com/android/repository/commandlinetools-linux-8512546_latest.zip
!mkdir -p $HOME/Android/Sdk/cmdline-tools
!unzip -q commandlinetools-linux-8512546_latest.zip -d $HOME/Android/Sdk/cmdline-tools
!mv $HOME/Android/Sdk/cmdline-tools/cmdline-tools $HOME/Android/Sdk/cmdline-tools/latest

# Set environment variables
import os
os.environ["ANDROID_SDK_ROOT"] = f"{os.path.expanduser('~')}/Android/Sdk"
os.environ["PATH"] += f":{os.path.expanduser('~')}/Android/Sdk/platform-tools:{os.path.expanduser('~')}/Android/Sdk/cmdline-tools/latest/bin"

# Accept licenses and install platform tools and NDK
!yes | sdkmanager --licenses
!sdkmanager "platform-tools" "platforms;android-33" "ndk;25.2.9519653"

# Step 2: Install Python Dependencies
!pip install --upgrade pip
!pip install flet cython
!pip install git+https://github.com/flet-dev/python-for-android.git@3.11.6

# Step 3: Create a simple Flet app (Hello World)
hello_world_code = """
import flet as ft

def main(page: ft.Page):
    page.add(ft.Text("Hello, World!"))

ft.app(target=main)
"""

# Save the hello world code to a Python file
with open("main.py", "w") as file:
    file.write(hello_world_code)

# Step 4: Build the APK using p4a and Flet
# Create APK distribution with p4a
!p4a create --requirements python3,flet --arch arm64-v8a --arch armeabi-v7a --arch x86_64 --sdk-dir $ANDROID_SDK_ROOT --ndk-dir $ANDROID_SDK_ROOT/ndk/25.2.9519653 --dist-name my_app_dist

# Build APK using Flet command
!flet build apk --template gh:flet-dev/flet-build-template -vv

# Step 5: Verify the APK output
!echo "APK files generated in ./build/apk:"
!ls ./build/apk

```

**Comment by ndonkoHenri** [comment]
@gunaNeelamegam upgrade to the latest prerelease: `pip install flet --pre -U`

**Comment by gunaNeelamegam** [comment]
> i Try to many way to build apk but its not working 
> ``` `flutter` command is not available in PATH. Install Flutter SDK.  ```
> 
> first i try to build my ubuntu not work same error so now i try in google cloab using this command but same error 
> ``` `flutter` command is not available in PATH. Install Flutter SDK.  ```
> 
> ```
> 
> # Step 1: Install Java, Android SDK, and NDK
> !apt-get update -y
> !apt-get install -y openjdk-11-jdk
> 
> # Download and set up Android SDK and NDK
> !wget https://dl.google.com/android/repository/commandlinetools-linux-8512546_latest.zip
> !mkdir -p $HOME/Android/Sdk/cmdline-tools
> !unzip -q commandlinetools-linux-8512546_latest.zip -d $HOME/Android/Sdk/cmdline-tools
> !mv $HOME/Android/Sdk/cmdline-tools/cmdline-tools $HOME/Android/Sdk/cmdline-tools/latest
> 
> # Set environment variables
> import os
> os.environ["ANDROID_SDK_ROOT"] = f"{os.path.expanduser('~')}/Android/Sdk"
> os.environ["PATH"] += f":{os.path.expanduser('~')}/Android/Sdk/platform-tools:{os.path.expanduser('~')}/Android/Sdk/cmdline-tools/latest/bin"
> 
> # Accept licenses and install platform tools and NDK
> !yes | sdkmanager --licenses
> !sdkmanager "platform-tools" "platforms;android-33" "ndk;25.2.9519653"
> 
> # Step 2: Install Python Dependencies
> !pip install --upgrade pip
> !pip install flet cython
> !pip install git+https://github.com/flet-dev/python-for-android.git@3.11.6
> 
> # Step 3: Create a simple Flet app (Hello World)
> hello_world_code = """
> import flet as ft
> 
> def main(page: ft.Page):
>     page.add(ft.Text("Hello, World!"))
> 
> ft.app(target=main)
> """
> 
> # Save the hello world code to a Python file
> with open("main.py", "w") as file:
>     file.write(hello_world_code)
> 
> # Step 4: Build the APK using p4a and Flet
> # Create APK distribution with p4a
> !p4a create --requirements python3,flet --arch arm64-v8a --arch armeabi-v7a --arch x86_64 --sdk-dir $ANDROID_SDK_ROOT --ndk-dir $ANDROID_SDK_ROOT/ndk/25.2.9519653 --dist-name my_app_dist
> 
> # Build APK using Flet command
> !flet build apk --template gh:flet-dev/flet-build-template -vv
> 
> # Step 5: Verify the APK output
> !echo "APK files generated in ./build/apk:"
> !ls ./build/apk
> 
> ```
> 

@MominIqbal-1234 

Install flutter and add the flutter bin path in the .bashrc file.
